resources/jquery.ui/jquery.ui.autocomplete.js

```diff
@@ -88,8 +88,10 @@ $.widget( "ui.autocomplete", {
 					self.menu.select( event );
 					break;
 				case keyCode.ESCAPE:
-					self.element.val( self.term );
-					self.close( event );
+					if ( self.menu.element.is( ":visible" ) ) {
```

Comment:
I did it the same as it was done in v. 1.9 of jquery ui (https://github.com/jquery/jquery-ui/blob/1-9-stable/ui/jquery.ui.autocomplete.js#L114). I think it's safer to keep it as is.